CMakeLists.txt

```diff
@@ -68,8 +68,24 @@ endif()
 stlab_detect_thread_system(STLAB_DEFAULT_THREAD_SYSTEM)
 set(STLAB_THREAD_SYSTEM ${STLAB_DEFAULT_THREAD_SYSTEM} CACHE STRING "Thread system to use (win32|pthread|pthread-emscripten|pthread-apple|none)")
 
-option(STLAB_MINIMAL_TASK_POOL "Use the minimal task pool size (ON|OFF), implies `STLAB_TASK_SYSTEM=portable`" OFF)
-if(STLAB_MINIMAL_TASK_POOL)
+#
+# TODO (bmedina): When STLab v3.0 is released, remove the deprecated STLAB_MINIMAL_TASK_POOL option
+# and simplify the code below.
+#
+option(STLAB_MINIMAL_TASK_POOL "Deprecated: Use STLAB_TASK_POOL_SIZE=1." OFF)
```

Comment:
This should be a MAXIMUM - we don't want to be larger than the reported hardware_concurrency size.